pallets/ajuna-affiliate/src/lib.rs

```diff
@@ -0,0 +1,269 @@
+// Ajuna Node
+// Copyright (C) 2022 BlogaTech AG
+
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+#![cfg_attr(not(feature = "std"), no_std)]
+
+pub use pallet::*;
+
+#[cfg(test)]
+mod mock;
+
+#[cfg(test)]
+mod tests;
+
+pub mod traits;
+
+use frame_support::pallet_prelude::*;
+
+use traits::*;
+
+#[frame_support::pallet]
+pub mod pallet {
+	use super::*;
+	use sp_runtime::ArithmeticError;
+
+	pub type AffiliatedAccountsOf<T, I> =
+		BoundedVec<<T as frame_system::Config>::AccountId, <T as Config<I>>::AffiliateMaxLevel>;
+
+	pub type PayoutRuleOf<T, I> = PayoutRule<<T as Config<I>>::AffiliateMaxLevel>;
+
+	pub type AccountIdOf<T> = <T as frame_system::Config>::AccountId;
```

Comment:
This is a nitpick, but substrate has introduced the convention of using `For` instead of `Of` in these type aliases. I usually prefer aligning those practices to help people who are used to substrate jargon.